Userland/Libraries/LibCore/IODevice.cpp

```diff
@@ -275,15 +275,19 @@ bool IODevice::truncate(off_t size)
     return true;
 }
 
-bool IODevice::write(const u8* data, int size)
+ErrorOr<void> IODevice::write(const u8* data, int size)
```

Comment:
This improves the situation. Great!

… Can I bait you into another PR after this one? It would be nice if we could report the amount of bytes actually written, so something like `ErrorOr<size_t>` would be nice. This way, doing non-blocking I/O could be made much more reliable. Note that this is going to be a huge endeavor, though.

Reply:
If it wasn't just depreacted, definitely! I supose this fancy new Stream-thing might benifit from something similar though. If it doesn't already have it, that is - I haven't looked at it yet.